tests/unit/amo/pages/TestUserProfile.js

```diff
@@ -619,209 +567,214 @@ describe(__filename, () => {
     const user = getCurrentUser(store.getState().users);
 
     _setUserReviews({ store, userId: user.id });
 
-    const dispatchSpy = sinon.spy(store, 'dispatch');
-    const errorHandler = createStubErrorHandler();
-    const location = createFakeLocation({ query: { page: 1 } });
+    const dispatch = jest.spyOn(store, 'dispatch');
+    const location = getLocation({ userId, search: `?page=1` });
 
-    const root = renderUserProfile({
-      errorHandler,
+    renderUserProfile({
       location,
       params,
       store,
     });
 
-    dispatchSpy.resetHistory();
+    dispatch.mockClear();
 
     const newPage = '2';
 
-    root.setProps({
-      location: createFakeLocation({ query: { page: newPage } }),
-      params,
-    });
+    store.dispatch(
+      onLocationChanged({
+        pathname: getLocation({ userId }),
+        search: `?page=${newPage}`,
+      }),
+    );
 
-    sinon.assert.calledOnce(dispatchSpy);
-    sinon.assert.calledWith(
-      dispatchSpy,
+    expect(dispatch).toHaveBeenCalledWith(
       fetchUserReviews({
-        errorHandlerId: errorHandler.id,
+        errorHandlerId: createErrorHandlerId({ userId }),
         page: newPage,
         userId: user.id,
       }),
     );
   });
 
   it('fetches reviews if user is loaded', () => {
-    const { params, store } = signInUserWithUserId(100);
+    const userId = 100;
+    const { params } = signInUserWithProps({ userId });
     const user = getCurrentUser(store.getState().users);
 
-    const dispatchSpy = sinon.spy(store, 'dispatch');
-    const errorHandler = createStubErrorHandler();
+    const dispatch = jest.spyOn(store, 'dispatch');
 
     const page = 123;
-    const location = createFakeLocation({ query: { page } });
+    const location = getLocation({ userId, search: `?page=${page}` });
 
-    renderUserProfile({ errorHandler, location, params, store });
+    renderUserProfile({ location, params });
 
-    sinon.assert.calledWith(
-      dispatchSpy,
+    expect(dispatch).toHaveBeenCalledWith(
       fetchUserReviews({
-        errorHandlerId: errorHandler.id,
-        page,
+        errorHandlerId: createErrorHandlerId({ userId }),
+        page: `${page}`,
         userId: user.id,
       }),
     );
   });
 
   it('does not fetch reviews if already loaded', () => {
-    const { params, store } = signInUserWithUserId(100);
+    const { params } = signInUserWithProps({ userId: 100 });
     const user = getCurrentUser(store.getState().users);
 
     _setUserReviews({ store, userId: user.id });
 
-    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const dispatch = jest.spyOn(store, 'dispatch');
 
-    renderUserProfile({ params, store });
+    renderUserProfile({ params });
 
-    sinon.assert.neverCalledWithMatch(dispatchSpy, {
-      type: FETCH_REVIEWS,
-    });
+    expect(dispatch).not.toHaveBeenCalledWith(
+      expect.objectContaining({ 'type': FETCH_USER_REVIEWS }),
+    );
+    expect(dispatch).not.toHaveBeenCalledWith(
```

Comment:
I don't think we need this, but it looks like this one has been around for a long time, so we should check. See if you can get the test to fail by keeping this assertion in, but commenting out `_setUserReviews`, above. If this assertion does not fail in that case, then it's an invalid assertion (which I suspect it is).

Reply:
it doesn't fail (`FETCH_USER_REVIEWS` does though)